File: backend/time-tracker-service/src/main/java/de/hnnsb/timetrackerservice/categories/Category.java
package de.hnnsb.timetrackerservice.categories;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Entity
@Getter
@Setter
public class Category {
    @Id
    @GeneratedValue
    private UUID id;
    private String name;
    private String color;
    private String email;
}

File: web-app-offline/src/lib/model/category.ts
import { v4 as uuidv4 } from "uuid";

export class Category {
  id: string | undefined;
  name: string;
  color: string;

  constructor(name: string, color: string) {
    this.id = uuidv4();
    this.name = name;
    this.color = color;
  }
}
